tutorbackup/templates/backup/build/backup/backup_local_services.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+set -e
+mysqldump --all-databases --password=$MYSQL_ROOT_PASSWORD --user=$MYSQL_ROOT_USERNAME --host=$MYSQL_HOST --port=$MYSQL_PORT > /data/mysql_dump.sql
```

Comment:
Please add [the `--single-transaction` flag](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html#option_mysqldump_single-transaction) here so we can get a consistent database dump without locking the entire database.

We might also consider adding `--quick`, to accommodate large tables.

Reply:
I just checked, and an Ansible-managed Open edX Lilac system drops the following options into `/etc/mysql/conf.d/mysqldump.cnf`:

```
[mysqldump]                                                                                                                                                                                   
quick                                                                                                                                                                                        
quote-names                                                                                                                                                                                   
max_allowed_packet      = 16M   
```

So, maybe you'll want to add `--quote-names` and `--max-allowed-packet=16M` as well.

Reply:
OK